static/src/javascripts/boot.js

```diff
@@ -0,0 +1,71 @@
+import domready from 'domready';
+import raven from 'lib/raven';
+import bootStandard from 'bootstraps/standard/main';
+import config from 'lib/config';
+import userTiming from 'lib/user-timing';
+import capturePerfTimings from 'lib/capture-perf-timings';
+
+// let webpack know where to get files from
+// __webpack_public_path__ is a special webpack variable
+// https://webpack.js.org/guides/public-path/#set-value-on-the-fly
+// eslint-disable-next-line camelcase,no-undef
+__webpack_public_path__ = `${config.page.assetsPath}javascripts/`;
+
+domready(() => {
+    // 1. boot standard, always
+    userTiming.mark('standard boot');
+    bootStandard();
+
+    // 2. once standard is done, next is commercial
+    if (config.switches.commercial) {
+        if (config.page.isDev) {
+            window.guardian.adBlockers.onDetect.push(isInUse => {
+                const needsMessage = isInUse &&
+                    window.console &&
+                    window.console.warn;
+                const message = 'Do you have an adblocker enabled? Commercial features might fail to run, or throw exceptions.';
+                if (needsMessage) {
+                    window.console.warn(message);
+                }
+            });
+        }
+
+        userTiming.mark('commercial request');
+        require.ensure(
+            [],
+            require => {
+                raven.context({ tags: { feature: 'commercial' } }, () => {
+                    userTiming.mark('commercial boot');
+                    require('bootstraps/commercial').init().then(() => {
```

Comment:
This is getting a little pyramid-y because of the `require.ensure` then `require` stuff, do you think it could do with extracting the functions so the `then`s are clearer?